From bc13f64fbabb9e8de17486fd4e7abff650a60ba7 Mon Sep 17 00:00:00 2001
From: Rico Ganahl <rico.ganahl@bytesatwork.ch>
Date: Tue, 1 Mar 2022 13:49:53 +0100
Subject: [PATCH] bytesatwork: bawconfig: adapt PCB, RAM and flash enums

Add available products to PCB enum.
RAM enum will be used to setup the RAM timings.
Additionally, use generic EMMC enum names.

Signed-off-by: Rico Ganahl <rico.ganahl@bytesatwork.ch>
(cherry picked from commit 8aa81503bc22f360588e19c831f3bc74e60f69d1)
---
 board/bytesatwork/baw_config/baw_config.c | 28 +++++++++++++++--------
 board/bytesatwork/baw_config/baw_config.h | 18 +++++++++++----
 2 files changed, 33 insertions(+), 13 deletions(-)

diff --git a/board/bytesatwork/baw_config/baw_config.c b/board/bytesatwork/baw_config/baw_config.c
index b28f302342..4dfa4fd520 100644
--- a/board/bytesatwork/baw_config/baw_config.c
+++ b/board/bytesatwork/baw_config/baw_config.c
@@ -6,11 +6,15 @@
 #include "baw_config.h"
 
 const struct baw_config_name_pair baw_config_pcb_name[] = {
-	{ M2_PCB_REV_00, "Rev.00" },
-	{ M2_PCB_REV_01, "Rev.01" },
-	{ M2_PCB_REV_02, "Rev.02" },
-	{ M2_PCB_REV_03, "Rev.03" },
-	{ M2_PCB_REV_04, "Rev.04" },
+	{ M2_PCB_REV_00, "M2 Rev.00" },
+	{ M2_PCB_REV_01, "M2 Rev.01" },
+	{ M2_PCB_REV_02, "M2 Rev.02" },
+	{ M2_PCB_REV_03, "M2 Rev.03" },
+	{ M2_PCB_REV_04, "M2 Rev.04" },
+	{ M5_PCB_REV_1_0, "M5 Rev 1.0" },
+	{ M5_PCB_REV_1_2, "M5 Rev 1.2" },
+	{ M6_PCB_REV_1_0, "M6 Rev 1.0" },
+	{ M6_PCB_REV_1_1, "M6 Rev 1.1" },
 	{ -1, NULL },
 };
 
@@ -20,16 +24,22 @@ const struct baw_config_name_pair baw_config_ram_name[] = {
 	{ M2_RAM_K4B2G1646QBCK0,   "K4B2G1646QBCK0 256 MB DDR3 @ 400Mhz" },
 	{ M2_RAM_K4B4G1646DBIK0,   "K4B4G1646DBIK0 512 MB DDR3 @ 400MHz" },
 	{ M2_RAM_MT41K512M16HA107, "MT41K512M16HA107 1024MB DDR3 @ 303MHz" },
+	{ M6_RAM_MT53E128M32D2DS_053, "MT53E128M32D2DS-053 512 MB LPDDR4 @ 1500 MHz" },
+	{ M6_RAM_MT53E256M32D2DS_053, "MT53E256M32D2DS-053 1 GB LPDDR4 @ 1500 MHz" },
+	{ M6_RAM_MT53E384M32D2DS_053, "MT53E384M32D2DS-053 1.5 GB LPDDR4 @ 1500 MHz" },
+	{ M6_RAM_MT53D512M32D2DS_053, "MT53D512M32D2DS-053 2 GB LPDDR4 @ 1500 MHz" },
+	{ M6_RAM_MT53E768M32D4DT_053, "MT53E768M32D4DT-053 3 GB LPDDR4 @ 1500 MHz" },
+	{ M6_RAM_MT53D1024M32D4DT_053, "MT53D1024M32D4DT-053 4 GB LPDDR4 @ 1500 MHz" },
 	{ -1, NULL },
 };
 
 const struct baw_config_name_pair baw_config_flash_name[] = {
 	{ M2_NAND_2GBIT, "Nand 2 GBit" },
 	{ M2_NAND_4GBIT, "Nand 4 GBit" },
-	{ M2_EMMC_4GB, "EMMC 4 GB" },
-	{ M2_EMMC_8GB, "EMMC 8 GB" },
-	{ M2_EMMC_16GB, "EMMC 16 GB" },
-	{ M2_EMMC_32GB, "EMMC 32 GB" },
+	{ EMMC_4GB, "EMMC 4 GB" },
+	{ EMMC_8GB, "EMMC 8 GB" },
+	{ EMMC_16GB, "EMMC 16 GB" },
+	{ EMMC_32GB, "EMMC 32 GB" },
 	{ -1, NULL },
 };
 
diff --git a/board/bytesatwork/baw_config/baw_config.h b/board/bytesatwork/baw_config/baw_config.h
index 0bc76496b5..fefdfd5405 100644
--- a/board/bytesatwork/baw_config/baw_config.h
+++ b/board/bytesatwork/baw_config/baw_config.h
@@ -15,6 +15,10 @@ typedef enum {
 	M2_PCB_REV_02 = 2,
 	M2_PCB_REV_03 = 3,
 	M2_PCB_REV_04 = 4,
+	M5_PCB_REV_1_0 = 5,
+	M5_PCB_REV_1_2 = 6,
+	M6_PCB_REV_1_0 = 7,
+	M6_PCB_REV_1_1 = 8,
 } baw_config_pcb_t;
 
 typedef enum {
@@ -23,15 +27,21 @@ typedef enum {
 	M2_RAM_K4B2G1646QBCK0   = 3,
 	M2_RAM_K4B4G1646DBIK0   = 4,
 	M2_RAM_MT41K512M16HA107 = 5,
+	M6_RAM_MT53E128M32D2DS_053 = 6,
+	M6_RAM_MT53E256M32D2DS_053 = 7,
+	M6_RAM_MT53E384M32D2DS_053 = 8,
+	M6_RAM_MT53D512M32D2DS_053 = 9,
+	M6_RAM_MT53E768M32D4DT_053 = 10,
+	M6_RAM_MT53D1024M32D4DT_053 = 11,
 } baw_config_ram_t;
 
 typedef enum {
 	M2_NAND_2GBIT  = 0,
 	M2_NAND_4GBIT  = 1,
-	M2_EMMC_4GB  = 2,
-	M2_EMMC_8GB  = 3,
-	M2_EMMC_16GB = 4,
-	M2_EMMC_32GB = 5,
+	EMMC_4GB  = 2,
+	EMMC_8GB  = 3,
+	EMMC_16GB = 4,
+	EMMC_32GB = 5,
 } baw_config_flash_t;
 
 struct baw_config {
-- 
2.39.2

